cfme/tests/insights/test_rhi_inventory.py

```diff
@@ -0,0 +1,154 @@
+from cfme.markers.env_markers.provider import ONE_PER_VERSION
+from cfme.utils import conf, ssh
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.conf import credentials
+from cfme.utils.hosts import setup_host_creds
+from cfme.utils.log import logger
+from cfme.utils.wait import wait_for, wait_for_decorator
+
+
+pytestmark = [
+    test_requirements.insights,
+    pytest.mark.meta(server_roles="+smartproxy +smartstate"),
+    pytest.mark.provider([VMwareProvider], selector=ONE_PER_VERSION, required_fields=[
+        ['vm_analysis_new', 'provisioning', 'vlan'],
+        ['vm_analysis_new', 'provisioning', 'host'],
+        ['vm_analysis_new', 'provisioning', 'datastore']]),
+    pytest.mark.usefixtures('setup_provider')
+]
+
+
+@pytest.fixture
+def register_appliance(request, appliance, reg_method='rhsm'):
+    red_hat_updates = RedHatUpdates(service='rhsm', url='subscription.rhn.redhat.com',
+                                    username=conf.credentials[reg_method]['username'],
+                                    password=conf.credentials[reg_method]['password'])
+
+    if not red_hat_updates.is_registered:
+        red_hat_updates.update_registration()
+
+        red_hat_updates.register_appliances()
+        wait_for(
+            func=red_hat_updates.is_registering,
+            func_args=[appliance.server.name],
+            delay=10,
+            num_sec=300,
+            fail_func=red_hat_updates.refresh
+        )
+
+        wait_for(
+            func=red_hat_updates.is_registered,
+            func_args=[appliance.server.name],
+            delay=20,
+            num_sec=500,
+            fail_func=red_hat_updates.refresh
+        )
+    request.addfinalizer(appliance.unregister)
+
+
+@pytest.fixture(scope="module")
+def ssa_analysis_profile():
+    collected_files = []
+    for file in ["/etc/hosts", "/etc/passwd", "/etc/redhat-access-insights/machine-id"]:
+        collected_files.append({"Name": file, "Collect Contents?": True})
+
+    analysis_profile_name = 'default'
+    analysis_profile = AnalysisProfile(
+        name=analysis_profile_name,
+        description=analysis_profile_name,
+        profile_type=AnalysisProfile.VM_TYPE,
+        categories=["System"],
+        files=collected_files
+    )
+    analysis_profile.create()
+    yield analysis_profile
+    analysis_profile.delete()
+
+
+@pytest.fixture
+def vm(request, provider, ssa_analysis_profile, register_appliance):
+    """ Fixture to provision instance on the provider """
+    # TODO use fauxfactory instead of random_vms
+    vm_name = 'test-rhi-{}'.format(fauxfactory.gen_alphanumeric())
+    vm_data = provider.data.vm_analysis_new
+    provisioning_data = vm_data.provisioning
+    collection = provider.appliance.provider_based_collection(provider)
+    vm_obj = collection.instantiate(vm_name,
+                                    provider,
+                                    template_name=vm_data['vms']['rhel74']['image'])
```

Comment:
**QUESTION**
Why do you use that template instead of `full_template`?

Reply:
For RHI testing RHEL template is required and till Ruslana adds more templates options in yamls.
Now I can use precise one there.